SCClassLibrary/Common/Math/Rational.sc

```diff
@@ -0,0 +1,194 @@
+Rational : Number {
+	var <>numerator, <>denominator;
+
+	*new { arg numerator, denominator = 1;
+		^super.new.initRational(numerator, denominator);
+	}
+
+	*newFromString { arg thisString;
+		^thisString.asRational;
+	}
+
+	initRational { arg thisNumerator, thisDenominator;
+		var factor;
+		if(thisNumerator.isString) {
+			^thisNumerator.asRational
+		} {
+			if(thisNumerator.isKindOf(Rational) or: thisDenominator.isKindOf(Rational)) {
+				^(thisNumerator.asRational / thisDenominator.asRational)
+			} {
+				if(thisNumerator.frac == 0 and: thisDenominator.frac == 0) {
+					if(thisDenominator == 0) {
+						this.numerator_(0);
+						this.denominator_(0);
+					} {
+						factor = gcd(thisNumerator.abs.asInteger, thisDenominator.abs.asInteger);
+						(thisDenominator < 0).if({factor = factor.neg});
+						(thisNumerator   < 0).if({factor = factor.neg});
+						this.numerator_((((thisNumerator/factor).abs) * factor.sign).asInteger);
+						this.denominator_((thisDenominator/factor).abs.asInteger);
+					};
+				} {
+					^(thisNumerator / thisDenominator).asRational
+				};
+			};
+		};
+	}
+	
+	performBinaryOpOnSimpleNumber { arg aSelector, aNumber, adverb;
+		^aNumber.asRational.perform(aSelector, this, adverb)
+	}
+	
+	isRational { ^true }
+	
+	%/ { arg aNumber; ^Rational(this, aNumber) }
+	
+	asRational { ^this }
+	
+	as { arg aSimilarClass; ^aSimilarClass.new(this.numerator, this.denominator) }
+	
+	asArray { ^[this.numerator, this.denominator]}
+	
+	asFloat { ^(this.numerator / this.denominator)}
+	
+	asInteger { ^((this.numerator / this.denominator).asInteger) }
+	
+	asInt { ^this.asInteger }
+	
+	reciprocal { ^Rational(this.denominator, this.numerator) }
+	
+	printOn { arg stream;
+		stream << "Rational( " << numerator << ", " << denominator << " )";
+	}
+	
+	hash { ^numerator.hash << 2 bitXor: denominator.hash } 
+	
+	+ { arg aNumber, adverb;
+		if(aNumber.isNumber) {
+			^this.species.new(
+				(this.numerator * aNumber.denominator) + (this.denominator * aNumber.numerator),
+				this.denominator * aNumber.denominator)
+		} {
+			^aNumber.performBinaryOpOnRational('+', this, adverb)
+		};
+	}
+	
+	- { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^this.species.new(
+				(this.numerator * aNumber.denominator) - (this.denominator * aNumber.numerator),
+				this.denominator * aNumber.denominator )
+		}  {
+			^aNumber.performBinaryOpOnRational('-', this, adverb)
+		};	
+	}
+	
+	* { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
```

Comment:
you can put the return `^` on the outside of the `if` to avoid multiple ones. Not a big issue, but I prefer it in many cases because it conveys that you are returning something new in any case.

Apart from this, the above comment counts here, too, no `isKindOf` should be necessary.